erts/emulator/drivers/common/efile_drv.c

```diff
@@ -1290,6 +1340,11 @@ static void invoke_read_file(void *data)
     }
     /* Invariant: d->c.read_file.size >= d->c.read_file.offset */
     
+    if (d->c.read_file.size != 0) {
```

Comment:
97e33f9 accidentally inverted this condition, it should be `d->c.read_file.size == 0`

`read_file_zero_size` sets `d->again` to 0 and doesn't close the file on its own, so jumping to `chop_done` will leak. `goto close;` instead?